### PR TITLE
newrelic_deployment: add app_name_exact_match parameter

### PR DESCRIPTION
**Pull request by matiasba**
##### SUMMARY
Adds the optional **app_name_exact_match** flag to the newrelic_deployment module.
This flag ensures that the exact app id is obtained from the newrelic api and not the first result if there is also a parcial match.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
newrelic_deployment

##### ADDITIONAL INFORMATION
In my opinion this should be the default behavior but I set it as optional so that it doesn't break anyone's usage.

### CODE REVIEW

**Comment by matiasba** [comment]
ready_for_review

File: changelogs/fragments/7355-newrelic-deployment-add-exact-name.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "newrelic_deployment - add option ``app_name_exact_match``, which filters results for the exact app_name provided"
```
Comment (felixfontein) [inline changelogs/fragments/7355-newrelic-deployment-add-exact-name.yml:2]:
```suggestion
  - "newrelic_deployment - add option ``app_name_exact_match``, which filters results for the exact app_name provided (https://github.com/ansible-collections/community.general/pull/7355)."
```

File: plugins/modules/newrelic_deployment.py [hunk 2]
```diff
@@ -68,6 +68,12 @@
     required: false
     default: true
     type: bool
+  app_name_exact_match:
+    type: bool
+    description:
+      - If this flag is set to True then the app id lookup by name would only work for an exact match.
```
Comment (felixfontein) [inline plugins/modules/newrelic_deployment.py:74]:
```suggestion
      - If this flag is set to V(true) then the app id lookup by name would only work for an exact match.
```

File: plugins/modules/newrelic_deployment.py [hunk 3]
```diff
@@ -68,6 +68,12 @@
...
+        If set to false it returns the first result
```
Comment (felixfontein) [inline plugins/modules/newrelic_deployment.py:75]:
```suggestion
        If set to V(false) it returns the first result.
```

File: plugins/modules/newrelic_deployment.py [hunk 4]
```diff
@@ -68,6 +68,12 @@
...
+    required: false
```
Comment (felixfontein) [inline plugins/modules/newrelic_deployment.py:76]:
```suggestion
    required: false
    default: false
    version_added: 7.5.0
```

File: plugins/modules/newrelic_deployment.py [hunk 5]
```diff
@@ -102,6 +108,7 @@ def main():
             revision=dict(required=True),
             user=dict(required=False),
             validate_certs=dict(default=True, type='bool'),
+            app_name_exact_match=dict(required=False, type='bool')
```
Comment (felixfontein) [inline plugins/modules/newrelic_deployment.py:111]:
```suggestion
            app_name_exact_match=dict(required=False, type='bool', default=False),
```

File: plugins/modules/newrelic_deployment.py [hunk 6]
```diff
@@ -102,6 +108,7 @@ def main():
...
         ),
         required_one_of=[['app_name', 'application_id']],
```
Comment (felixfontein) [inline plugins/modules/newrelic_deployment.py:113]:
```suggestion
        required_one_of=[['app_name', 'application_id']],
        required_if=[('app_name_exact_match', True, ['app_name'])],
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by matiasba** [comment]
@felixfontein All suggestions applied!

File: plugins/modules/newrelic_deployment.py [hunk 7]
```diff
@@ -68,6 +68,14 @@
     required: false
     default: true
     type: bool
+  app_name_exact_match:
+    type: bool
+    description:
+      - If this flag is set to V(true) then the app id lookup by name would only work for an exact match.
```
Comment (felixfontein) [inline plugins/modules/newrelic_deployment.py:74]:
```suggestion
      - If this flag is set to V(true) then the application ID lookup by name would only work for an exact match.
```

**Review by felixfontein — COMMENTED** [review]
Sorry, I missed this one.

If nobody objects, I'll merge this at the end of the weekend.

**Comment by felixfontein** [comment]
@matiasba thanks for your contribution!

**Pull request merged**